### ISSUE TITLE
Händler Upgrades

### ISSUE THREAD

**Issue by Plushtoast** [Plushtoast/dsa5-foundryVTT#736]
Der Händler hat ja nun geraume Zeit und Testzeit hinter sich, damit wäre vlt. mal ein Upgrade sinnvoll.
Bitte fügt Ideen und Änderungen hier an

- [x] Händler sollte keine Talismane verkaufen (bzw Preis über gewisser Schwelle)
- [x] Geweihte Gegenstände generell nicht verkaufen
- [x] Zufallskategorien nach Subkategorien der Ausrüstungsgegenstände erlauben (z.b. Dienstleistungen)
- [x] Zurücksetzen löscht keine Pflanzen und Munition
- [x] Kauffaktor von 0 soll kostenlose Übergabe ermöglichen
- [x] einen Preisfaktor für jeden Spieler getrennt eingeben kann.
- [ ] Handelsprobe?

Update v2:

- [x] Button to give access to all players at once

Update v3:
- [x] Optionale Chatnachricht bei Itembesitzer wechsel
- [x] Merchant sheet für charaktere & creatures
- [x] Optionale option um token zu löschen, wenn inventar leer ist
- [x] Drop items to ground
- [x] Player trade window
- [ ] (Maybe) share money equally button
- [x] Beliebige Handelspartner
- [x] disable item from being traded
- [x] Detect any owned actor next to merchant (DPS)
- [x] Increase amount of all items by once (# button)
- [x] Random gear with quantity value (also in workshop)
- [x] Macro transform to loot
- [x] Detailed Macro transform to loot
- [x] Confirm player to player transaction somehow

**Comment by Nellkir** [comment]
Da man mit einer Handel-Probe den Preisfaktor des Händlers beeinflussen kann, wäre es ganz nett, wenn man optional noch einen Preisfaktor für jeden Spieler getrennt eingeben kann.

**Comment by mckee14** [comment]
da würde sich eine vergleichsprobe anbieten zwisch spieler und händler,
wobei die NettoQS den preis für diesen spieler bzw diesen handelsvorgang
beeinflusst.

dann dürfte aber nur jeweils 1 spieler den händler öffnen oder mit ihm
interagieren können.

Nellkir ***@***.***> schrieb am Di., 7. Dez. 2021, 21:12:

**Comment by TheFrazer** [comment]
@mckee14 Die offizielle Regel (GRW. 208) sagt ja genau das ;)
> Auf dem Markt feilschen: Vergleichsprobe (Handel [Feilschen] gegen Handel [Feilschen]) (Nachdem die QS des Verlierers von den QS des Gewinners abgezogen wurden, steigt oder fällt Kaufpreis um die verbliebenen QS x 10 in Prozent nach Wahl des Siegers, maximal jedoch um 50 %.)

**Comment by mckee14** [comment]
ja klar, deswegen habe ich das ja eingeworfen.

es macht daher wenig sinn, im Händler oder char einen festen faktor zu
hinterlegen. der ist ja bei jedem handeln potentiell anders.

zudem gibt es vorteile und SF, die das beeinflussen.

daher wäre es praktischer, einen ablauf zu designen, der dies alles
berücksichtigt. der auslöser dafür dürfte dann eine handeln-probe sein.



Fritz ***@***.***> schrieb am Mi., 8. Dez. 2021, 12:16:

**Comment by TheFrazer** [comment]
Mich hat hieran aber immer massiv gestört, dass der Händler dir im Prinzip einen Preis sagt, du feilschst und der Händler plötzlich mehr will. Ich hab den Händler immer ein bisschen teurer als normal angesetzt und dann je nach netto QS den Preis gesenkt. Wenn der Händler gewonnen hat ist aber alles beim alten geblieben. Solche Extrawürste zu implementieren wird vermutlich nicht klappen, aber sollte ein solches Feilschen-System kommen, würde ich mich sehr für meine Variante stark machen wollen!

Zur Zufallsauswahl:
* Bei dem Thema mit den Talismanen würde ich noch hinzufügen, dass ich es besser fände, wenn die Händeler (über das Zufallsmenü) generell nichts aus der Kategorie "Geweiht" verkaufen.
* Ich würde mich freuen, wenn hier noch mehr Karegorien kommen würden. Zum Beispiel Dienstleistungen, Essen/Alkohol, Luxusgüter etc. 
* Vielleicht magisch auch als eigene Option? Wobei ich das als nicht so wichtig erachten würde. Bei den Zufallssachen muss man ja eh noch einmal nachgucken, dass alles passt.

Bugs:
* Zurücksetzten löscht keine Pflanzen und Munition
* Ein Kauffaktor von 0 wird als 1 gerechnet. Ich würde auch gerne Händler haben, die den Spielern nichts abnehmen (schenken ist mir hier dann egal). Klar, man kann den Händler auch einfach kein Geld geben...

**Comment by Nellkir** [comment]
Natürlich wäre es mega, wenn er die Handelsprobe automatisch erkennt, und in Faktoren umsetzt. Aber das wäre vielleicht doch recht komplex. Mein Vorschlag wäre da etwas simpler. Die Spieler können ihre Handelsvergleichsprobe mit dem Händler auswürfeln, und der Meister bekommt die Möglichkeit einfach den Faktor für betroffene Charaktere noch etwas anzupassen.
Der grundsätzliche Händlerfaktor sollte natürlich trotzdem erhalten bleiben.
![image](https://user-images.githubusercontent.com/82062265/145216133-a54c7a15-9ef3-4f99-aae2-0a696e4bf460.png)

**Comment by Plushtoast** [comment]
Das wäre schon machbar. Ist aber wahrscheinlich nur sinnvoll wenn der Wurf für die spieler blind ist, dass sie nicht sehen, dass sie auf einmal mehr zahlen... Oder so. Finde dreiste oder miserable handelsversuche dürfen den Händler schon beleidigen und den Preis hochtreiben

**Comment by Plushtoast** [comment]
So wie nellkir das gemalt hat würde ich es machen. Mit dem Faktor je Spieler der zusätzlich noch mit dem Basisfaktor multipliziert wird.
Und dann noch irgendwie die Probe auslösen...

**Comment by Nellkir** [comment]
Ich denke eine Preissteigerung kommt nur in Frage, wenn man nen Patzer würfelt oder dergleichen, nicht weil der Händler einen bessere handelsprobe gewürfelt hat (was vermutlich zu 80% der Fall sein wird)
Aber ich denke, das würde jede SL anders handhaben Geschmacktechnisch, weswegen ich den Faktor auch nicht unbedingt an eine Probe koppeln würde. ^^

**Comment by mckee14** [comment]
> Mich hat hieran aber immer massiv gestört, dass der Händler dir im Prinzip einen Preis sagt, du feilschst und der Händler plötzlich mehr will. Ich hab den Händler immer ein bisschen teurer als normal angesetzt und dann je nach netto QS den Preis gesenkt. Wenn der Händler gewonnen hat ist aber alles beim alten geblieben. Solche Extrawürste zu implementieren wird vermutlich nicht klappen, aber sollte ein solches Feilschen-System kommen, würde ich mich sehr für meine Variante stark machen wollen!
> 
> Zur Zufallsauswahl:
> 
> * Bei dem Thema mit den Talismanen würde ich noch hinzufügen, dass ich es besser fände, wenn die Händeler (über das Zufallsmenü) generell nichts aus der Kategorie "Geweiht" verkaufen.
> * Ich würde mich freuen, wenn hier noch mehr Karegorien kommen würden. Zum Beispiel Dienstleistungen, Essen/Alkohol, Luxusgüter etc.
> * Vielleicht magisch auch als eigene Option? Wobei ich das als nicht so wichtig erachten würde. Bei den Zufallssachen muss man ja eh noch einmal nachgucken, dass alles passt.
> 
> Bugs:
> 
> * Zurücksetzten löscht keine Pflanzen und Munition
> * Ein Kauffaktor von 0 wird als 1 gerechnet. Ich würde auch gerne Händler haben, die den Spielern nichts abnehmen (schenken ist mir hier dann egal). Klar, man kann den Händler auch einfach kein Geld geben...

Entweder man kauft etwas zu einem Preis ohne zu handeln. Dann kostet es X.

Oder man möchte etwas aushandeln, dann kostet X*Y. Dabei braucht man X für die Berechnung, der CHAR weiss davon aber nichts, der kennt nur Y. Dass man bei der Handelsprobe X benötigt, ist dem Regelmechanismus geschuldet.

kann also sein, dass man die Handelsprobe total verkackt, einen Gegenstand massiv überzahlt und sich dennoch freut wie Bolle, weil man so ein tolles Geschäft gemacht hat. Und das alles nur, weil man schlecht gehandelt hat.

Hat jetzt mit der möglichen Umsetzung nichts zu tun, wollte es bloss ein wenig Regeln und Fluff verheiraten :)

**Comment by TheFrazer** [comment]
Ja, ich hab ja gesagt, dass das so ist, wie ich es mitlerweile mache. Einfach weil ich es doof finde, dass dir ein Preis genannt wird, du versuchst den herunterzuhandeln und der dann teurer ist. Patzen (oder Krit vom Händler) mal ausgeschlossen.

Ich finde es so, wie Nellkir es skiziert hat, auch am besten!
